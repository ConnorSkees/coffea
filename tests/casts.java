public class casts {

    public static void main(String[] args) {
        int i = 1;
        float f = 2;
        double d = 3;
        long l = 4;

        int f_to_i = (int) f;
        int d_to_i = (int) d;
        int l_to_i = (int) l;

        float i_to_f = (float) i;
        float d_to_f = (float) d;
        float l_to_f = (float) l;

        double i_to_d = (double) i;
        double f_to_d = (double) f;
        double l_to_d = (double) l;

        long i_to_l = (long) i;
        long f_to_l = (long) f;
        long d_to_l = (long) d;

        byte b = (byte) i;
        char c = (char) i;
        short s = (short) i;

        // these are implicitly converted to ints and use i2d
        double b_to_d = (double) b;
        double c_to_d = (double) c;
        double s_to_d = (double) s;
    }
}