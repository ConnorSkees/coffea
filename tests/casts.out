public class casts {
  public static void main(String[] arg0) {
    int i1 = 1;
    float f2 = 2;
    double d3 = 3;
    long l5 = 4;
    int i7 = (int) f2;
    int i8 = (int) d3;
    int i9 = (int) l5;
    float f10 = (float) i1;
    float f11 = (float) d3;
    float f12 = (float) l5;
    double d13 = (double) i1;
    double d15 = (double) f2;
    double d17 = (double) l5;
    long l19 = (long) i1;
    long l21 = (long) f2;
    long l23 = (long) d3;
    byte by25 = (byte) i1;
    char ch26 = (char) i1;
    short sh27 = (short) i1;
    double d28 = (double) by25;
    double d30 = (double) ch26;
    double d32 = (double) sh27;
    return;
  }
}
